### PR TITLE
MII dynamic changes - Changing the listen port, address, SSL on a server or channel (unsupported)

### PR DESCRIPTION
**Pull request by xiancao**
These are negative tests for MII domain dynamic online updates.
- change listen port
- change listen address
- change SSL on a server

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3506/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3508/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3515/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 1]
```diff
@@ -549,33 +548,150 @@ public void testMiiChangeDomainName() {
     // Patch a running domain with introspectVersion.
     patchDomainResourceWithNewIntrospectVersion(domainUid, domainNamespace);
 
+    // Verifying introspector pod is created and failed
+    logger.info("verifying the introspector failed with the expected error msg");
+    String expectedErrorMsg = "Model in image online update failed because of forbidden changes";
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:553]: this can be moved to constants

**Review by vanajamukkara — APPROVED** [review]
LG

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java (hunk 1 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 2]
```diff
@@ -549,33 +550,158 @@ public void testMiiChangeDomainName() {
...
+
+    // Patch a running domain with introspectVersion.
+    patchDomainResourceWithNewIntrospectVersion(domainUid, domainNamespace);
+
+    // Verifying introspector pod is created, runs and deleted
+    verifyIntrospectorRuns();
+  }
+
+  /**
+   * Negative test: Changing the listen port of a server using mii dynamic update.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:581]: Can you describe details of the expected outcome in this case.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 3]
```diff
@@ -549,33 +550,158 @@ public void testMiiChangeDomainName() {
...
-  private boolean domainStatusConditionMsgContainsErrorMsg(Domain domain, String errorMsg) {
-    for (DomainCondition domainCondition : domain.getStatus().getConditions()) {
-      if (domainCondition.getType().equalsIgnoreCase("Failed")
-          && domainCondition.getMessage().contains(errorMsg)) {
-        return true;
-      }
-    }
-    return false;
+  /**
+   * Negative test: Changing the listen address of a server using mii dynamic update.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:623]: Describe the expected outcome

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 4]
```diff
@@ -549,33 +550,158 @@ public void testMiiChangeDomainName() {
...
+
+    // Patch a running domain with introspectVersion.
+    patchDomainResourceWithNewIntrospectVersion(domainUid, domainNamespace);
+
+    // Verifying introspector pod is created, runs and deleted
+    verifyIntrospectorRuns();
+  }
+
+  /**
+   * Negative test: Changing SSL setting of a server using mii dynamic update.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:665]: Describe the expected outcome.

**Review by anpanigr — COMMENTED** [review]
Can we catch Error message specific to a unsupported attribute instead of generic message "Model in image online update failed because of forbidden changes"

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java (hunk 2 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:585]: fix. 
To answer your question above, all the error msgs are the same.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java (hunk 3 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:631]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java (hunk 4 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:677]: fixed

**Review by anpanigr — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 5]
```diff
@@ -13,6 +13,7 @@
 import java.util.Map;
 import java.util.Set;
 
+import io.kubernetes.client.openapi.ApiException;
```
Comment (rjeberhard) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:16]: Need to update copyright date given that this is a new file. "2020," becomes "2020, 2021," and "Oracle Corporation" becomes just "Oracle".

Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:16]: fixed

**Pull request merged**